File: tools/parity/dispatch/tui-agent-startup.ts
// TS dispatch for the tui-agent-startup parity module. The shared TS bodies were
// DELETED (the Rust orca-agents core is the sole impl — napi in main, wasm in the
// renderer), so this adapter drives the SAME wasm through the single
// `tuiAgentStartupOp` boundary: the vectors' goldens pin that surface and the
// harness's TS-vs-Rust diff degenerates to wasm-vs-binary.
import { gitWasmOracle } from './orca-git-wasm-oracle'

export function dispatch(fn: string, input: unknown): unknown {
  return JSON.parse(gitWasmOracle().tuiAgentStartupOp(fn, JSON.stringify(input ?? null)))
}

File: rust/crates/orca-parity/src/modules/tui_agent_startup.rs
//! Parity dispatch for `orca_agents::tui_agent_startup` vs
//! `src/shared/tui-agent-startup.ts`. The JSON marshalling now lives in
//! `orca_agents::tui_agent_startup_json` so napi (main), wasm (renderer), and
//! this oracle share one boundary; this module just delegates.

use serde_json::Value;

pub fn dispatch(function: &str, input: &Value) -> Value {
    orca_agents::tui_agent_startup_json::dispatch(function, input)
}
